core/matrix/csr.cpp

```diff
@@ -536,6 +543,29 @@ bool Csr<ValueType, IndexType>::is_sorted_by_column_index() const
 }
 
 
+template <typename ValueType, typename IndexType>
+std::unique_ptr<Csr<ValueType, IndexType>>
+Csr<ValueType, IndexType>::create_submatrix(const gko::span& row_span,
+                                            const gko::span& column_span) const
+{
+    using Mat = Csr<ValueType, IndexType>;
+    auto exec = this->get_executor();
+    auto sub_mat_size = gko::dim<2>(row_span.length(), column_span.length());
+    Array<IndexType> row_ptrs(exec, row_span.length() + 1);
+    exec->run(csr::make_fill_array(
+        row_ptrs.get_data(), row_ptrs.get_num_elems(), zero<IndexType>()));
```

Comment:
Why is this necessary? The following step should write to each entry except the last one, which gets filled by the prefix_sum